src/public/services/UserService.ts

```diff
@@ -0,0 +1,14 @@
+import { User } from '../../types/api/user'
+
+/** Exported for testing */
+export const STORAGE_USER_KEY = 'user'
+
+/**
+ * Save logged in user to localStorage.
+ * May not be needed in React depending on implementation.
+ *
+ * @param user the user to save to local storage
+ */
+export const saveUserToLocalStorage = (user: User | null): void => {
+  localStorage.setItem(STORAGE_USER_KEY, JSON.stringify(user))
```

Comment:
if `user` is `null`, do we really want to set localstorage to the string `'null'`?

Reply:
seems it might be better to just use `clearUserFromLocalStorage` from part 4, which means the 'user' key is either set correctly or not, without this intermediate `'null'` state

Reply:
ok I now see that's already done

Reply:
I'll move them into this PR for completeness

Reply:
Oh clearing of user is not in this PR yet, I'll change the argument in this PR to only accept user

Reply:
Done in 5a03fa6